R/theme-elements.r

```diff
@@ -130,6 +131,38 @@ element_text <- function(family = NULL, face = NULL, colour = NULL,
   )
 }
 
+#' @param colour_1 geom accent colour 1,
+#'   typically a lighter version of colour
+#' @param colour_2 geom accent colour 2,
+#'   typically a bright colour used for geom_smooth et al.
+#' @param fill_1 geom accent fill colour, typically a darker version of fill
```

Comment:
Have y'all considered that there are a handful of different levels of (grayscale) fill in ggplot2 alone (likely for a reason)?

```r
> geoms <- mget(ls(asNamespace("ggplot2"), pattern = "^Geom[A-Z]"), asNamespace("ggplot2"))
> unique(unlist(lapply(geoms, function(g) g$default_aes$fill)))
[1] "grey20" "grey35" "white"  NA       "black"  "grey50" "grey60"
```

With that in mind, a more robust interface would be something like `theme(geom = element_geom(bg = "white", fg = "black", accent = "#3366FF", ...))` and hold the geom responsible for mixing the background and foreground colors. For example, `GeomRect$default_aes$fill`  currently defaults to `"gray35"` (i.e., interpolate 35% of the way from `fg` to `bg`), so the default could be:

```r
GeomRect$default_aes$fill = scales::colour_ramp(c(from_theme('bg'), from_theme('fg')), alpha = TRUE)(0.35)
``` 

By the way, I've been thinking a lot about this sort of stuff recently for the new [wip auto-theming feature coming to shiny](https://github.com/rstudio/shiny/pull/2740), and I'm thinking of abstracting out some of that functionality  into a self-contained package so it could be used anywhere. Towards that end, having this PR, as well as #3828 and #3833, would be hugely helpful in avoiding terrible hacks to reach that goal

Reply:
In one of my earlier attempts at suggesting this geom-defaults-in-themes feature I floated the idea of extending `rel()` beyond sizes; one could think of defining a few main colours  — [a bit like a css-variables model](https://razorltd.github.io/sasscolourfunctioncalculator/) — and have functions to calculate derivative shades (e.g desaturated, complement, lighter/darker, etc.)

(The css analogy isn't entirely unmotivated as it seems likely more and more use-cases will want to style ggplots consistently with a surrounding webpage (cf Joe's comment on Shiny, for instance))